Realm/Realm/Native/PrimitiveValue.cs

```diff
@@ -284,6 +312,24 @@ private struct LinkValue
             public readonly TableKey table_key;
         }
 
+        [StructLayout(LayoutKind.Sequential)]
+        private struct ListValue
```

Comment:
I have just removed them and left only one pointer. The only minor disadvantage with this is that now we have a void pointer in the wrapper instead of two typed ones (`List*`, ..)